cfme/common/__init__.py

```diff
@@ -365,6 +367,65 @@ def step(self, *args, **kwargs):
         self.prerequisite_view.toolbar.policy.item_select('Manage Policies')
 
 
+class CompareView(BaseLoggedInPage):
+    """generic class for compare views"""
+    title = Text('.//div[@id="center_div" or @id="main-content"]//h1')
+    comparison_table = Table(locator='//div[@id="compare-grid"]/table')
+
+    @View.nested
+    class toolbar(View):
+        all_attributes = Button(title="All attributes")
+        different_values_attributes = Button(title="Attributes with different values")
+        same_values_attributes = Button(title="Attributes with same values")
+        details_mode = Button(title="Details Mode")
+        exists_mode = Button(title="Exists Mode")
+        # add expanded view, compressed view buttons.
+        download = Dropdown('Download')
+        view_selector = View.nested(ItemsToolBarViewSelector)
+
+
+class ComparableMixin:
+    """
+    Mixin for comparing entities. Should be added to Collection class.
+    Constants:
+            DROPDOWN_TEXT: (str) the text in the Configuration dropdown for comparing entities
+            NAV_STRING: (str) used as second arg in call to navigate_to()
+            COMPARE_VIEW: (VIEW) view class to create and return
+    Usage:
+        If added to collection's class,
+
+        entity_coll = collection.all()[:2]
+        compare_view = collection.compare_entities(provider, entities_list=entity_coll)
+
+        will navigate to the correct view, select the items in entity_col (first two items returned
+        from all()), click on the compare dropdown item and verify the compare view is
+        displayed. It then returns the compare view.
+    """
+    DROPDOWN_TEXT = 'Compare Selected items'
+    NAV_STRING = 'All'
+    COMPARE_VIEW = CompareView
+
+    def compare_entities(self, provider, entities_list=None):
```

Comment:
Please remove provider as required arg here - its only used for creating the view, and you can call `create_view` on `self` here, as its provided by NavigatableMixin on the collections.